differential-dataflow/src/trace/implementations/mod.rs

```diff
@@ -483,6 +483,15 @@ pub mod containers {
 
         /// Reference to the element at this position.
         fn index(&self, index: usize) -> Self::ReadItem<'_>;
+
+        /// Reference to the element at this position, if it exists.
+        fn get(&self, index: usize) -> Option<Self::ReadItem<'_>> {
+            if index < self.len() {
+                Some(self.index(index))
+            }
+            else { None }
+        }
```

Comment:
Do we want to provide a default implementation? If it all looks like this, then maybe not, otherwise we could avoid accidental performance problems by not providing it.

Reply:
Mostly it provides an incremental deployment, vs breaking for everyone. Seems ok either way, but easier to merge this way.